pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -390,14 +392,22 @@ def append_fn(v: ResultMetric) -> None:
         apply_to_collection(list(self.values()), ResultMetric, append_fn)
         return o
 
-    @property
-    def batch_size(self) -> torch.Tensor:
-        # performance: cache the `batch_size` tensor instead of re-creating it
-        return self._batch_size
-
-    @batch_size.setter
-    def batch_size(self, value: int) -> None:
-        self._batch_size = torch.tensor(value, device=self.device)
+    def _extract_batch_size(self, batch_size: Optional[int], meta: _Metadata) -> int:
+        # check if we have extracted the batch size already
+        batch_size = batch_size or self.batch_size
+        if batch_size is not None:
+            return batch_size
+        batch_size = 1
+
+        # extract it
+        fx_validate = _FxValidator.functions.get(meta.fx.split(".")[0])
+        if meta.on_epoch and fx_validate is not None and (True in fx_validate["on_step"]) and meta.is_mean_reduction:
```

Comment:
why exactly do we need the `True in fx_validate["on_step"]` check?

Reply:
well, we need to ensure that batch_size should only be extracted for the cases where step-level logging is allowed. Also in other cases `results.batch` will be None too. We can avoid this check by allowing the call for `extract_batch_size(None)` and returning 1 from there, but it didn't seem ideal to me.